tests/e2e/e2e_util_test.go

```diff
@@ -0,0 +1,185 @@
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/ory/dockertest/v3/docker"
+)
+
+func (s *IntegrationTestSuite) connectIBCChains() {
+	s.T().Logf("connecting %s and %s chains via IBC", s.chainA.id, s.chainB.id)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	exec, err := s.dkrPool.Client.CreateExec(docker.CreateExecOptions{
+		Context:      ctx,
+		AttachStdout: true,
+		AttachStderr: true,
+		Container:    s.hermesResource.Container.ID,
+		User:         "root",
+		Cmd: []string{
+			"hermes",
+			"create",
+			"channel",
+			s.chainA.id,
+			s.chainB.id,
+			"--port-a=transfer",
+			"--port-b=transfer",
+		},
+	})
+	s.Require().NoError(err)
+
+	var (
+		outBuf bytes.Buffer
+		errBuf bytes.Buffer
+	)
+
+	err = s.dkrPool.Client.StartExec(exec.ID, docker.StartExecOptions{
+		Context:      ctx,
+		Detach:       false,
+		OutputStream: &outBuf,
+		ErrorStream:  &errBuf,
+	})
+	s.Require().NoErrorf(
+		err,
+		"failed connect chains; stdout: %s, stderr: %s", outBuf.String(), errBuf.String(),
+	)
+
+	s.Require().Containsf(
+		errBuf.String(),
+		"successfully opened init channel",
+		"failed to connect chains via IBC: %s", errBuf.String(),
+	)
+
+	s.T().Logf("connected %s and %s chains via IBC", s.chainA.id, s.chainB.id)
+}
+
+func (s *IntegrationTestSuite) sendIBC(srcChainID, dstChainID, recipient string, token sdk.Coin) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	s.T().Logf("sending %s from %s to %s (%s)", token, srcChainID, dstChainID, recipient)
+
+	exec, err := s.dkrPool.Client.CreateExec(docker.CreateExecOptions{
+		Context:      ctx,
+		AttachStdout: true,
+		AttachStderr: true,
+		Container:    s.hermesResource.Container.ID,
+		User:         "root",
+		Cmd: []string{
+			"hermes",
+			"tx",
+			"raw",
+			"ft-transfer",
+			dstChainID,
+			srcChainID,
+			"transfer",  // source chain port ID
+			"channel-0", // since only one connection/channel exists, assume 0
+			token.Amount.String(),
+			fmt.Sprintf("--denom=%s", token.Denom),
+			fmt.Sprintf("--receiver=%s", recipient),
+			"--timeout-height-offset=1000",
+		},
+	})
+	s.Require().NoError(err)
+
+	var (
+		outBuf bytes.Buffer
+		errBuf bytes.Buffer
+	)
+
+	err = s.dkrPool.Client.StartExec(exec.ID, docker.StartExecOptions{
+		Context:      ctx,
+		Detach:       false,
+		OutputStream: &outBuf,
+		ErrorStream:  &errBuf,
+	})
+	s.Require().NoErrorf(
+		err,
+		"failed to send IBC tokens; stdout: %s, stderr: %s", outBuf.String(), errBuf.String(),
+	)
+
+	s.T().Log("successfully sent IBC tokens")
+}
+
+func queryGaiaTx(endpoint, txHash string) error {
+	resp, err := http.Get(fmt.Sprintf("%s/cosmos/tx/v1beta1/txs/%s", endpoint, txHash))
```

Comment:
A question really, should we interact with the chains over somthing other than rest?
essentially seeing as not all types of query/commands are available via rest, maybe rpc/grpc is a better approach in this case?